File: Sorting/Bubble_sort.py
# writing a code for the bubble sort algorithm
# the bubble sort algorithm is a sorting algorithm that compares two adjacent elements and swaps them if they are not in the intended order.
# swap the element inpairs such that the smallest element is at the beginning of the list.
# when reach at end of the list,start over agin.
# stop when the swap is no more needed.
# the bubble sort algorithm is not efficient for large lists.
def bubble_sort(L):
    swap = False
    while not swap:
        print('bubble sort:' + str(L))
        swap = True
        for j in range(1,len(L)):
            if L[j-1] > L[j]:
                swap = False
                temp = L[j]
                L[j] = L[j-1]
                L[j-1] = temp

testlist = [1,3,5,7,2,6,25,18,13]
bubble_sort(testlist)
print(testlist)

File: Sorting/BubbleSort/BubbleSort.cpp
// writing an code for the bubble sort
#include <iostream>
#include<bits/stdc++.h>
using namespace std;
// bubble sort function

// writing the swap function
void swap(int *a,int *b){
    int temp = *a;
    *a = *b;
    *b = temp;
}

void BubbleSort(int arr[],int n){
    for(int i =0;i<n;i++){
        for(int j =0;j<n-1;j++){
            if(arr[j]>arr[j+1]){
                swap(arr[j],arr[j+1]);

            }

        }

    }
}
// print array function
void printArray(int arr[],int n){
    for(int i =0;i<n;i++){
        cout<<"arr["<<i<<"]"<<" "<<arr[i]<<endl;
    }
}

int main(){
    int n;
    int arr[] = {4,6,8,2,1,9,3,5,7};
    n = sizeof(arr)/sizeof(arr[0]);
    for(int i =0;i<n;i++){
        cin>>arr[i];
    }
    BubbleSort(arr,n);
    printArray(arr,n);
    return 0;

}